int64 data
---
int64 res
